File: JavaScript/1090. Largest Values From Labels.js
/**
 * @param {number[]} values
 * @param {number[]} labels
 * @param {number} numWanted
 * @param {number} useLimit
 * @return {number}
 */
var largestValsFromLabels = function(values, labels, numWanted, useLimit) {
    var arr = [];
    var mp = new Map();
    var n = values.length;
    var res = 0;
    for(var i=0; i<n; i++){
        arr.push([values[i], labels[i]]);
    }
    arr.sort((a, b) => a[0] - b[0]);
    for(var i=0; i<n; i++){
        var temp = arr.pop();
        var [value, label] = temp;
        if(!mp.has(label)){
            mp.set(label, 0);
        }
        mp.set(label, mp.get(label) + 1);
        if(numWanted!=0 && mp.get(label) <= useLimit){
            res += value;
            numWanted--;
        }
    }
    return res;
};

File: C++/1090. Largest Values From Labels.cpp
class Solution {
public:
    int largestValsFromLabels(vector<int>& values, vector<int>& labels, int numWanted, int useLimit) {
        vector<pair<int, int>> v;
        map<int, int> mp;
        int n = values.size();
        int res = 0;
        for(int i=0; i<n; i++){
            v.push_back({values[i], labels[i]});
        }
        sort(v.begin(),v.end(),[](const auto& a, const auto& b){return a.first > b.first;});
        for(int i=0; i<n; i++){
            mp[v[i].second]++;
            if(mp.find(v[i].second) != mp.end() && numWanted && mp[v[i].second] <= useLimit){
                res += v[i].first;
                numWanted--;
            }
        }
        return res;
    }
};
